yletunnus/tests/test_backend.py

```diff
@@ -91,18 +96,40 @@ def authorize_body(self, request, url, headers):
 
     def auth_handlers(self, start_url):
         target_url = super().auth_handlers(start_url)
+        if self.strategy.setting('YLETUNNUS_ENVIRONMENT') == 'production':
+            assert start_url.startswith('https://auth.api.yle.fi/authorize')
+        elif self.strategy.setting('YLETUNNUS_ENVIRONMENT') == 'test':
+            assert start_url.startswith('https://auth.api-test.yle.fi/authorize')
         self.complete_url = target_url
         HTTPretty.register_uri(HTTPretty.GET, start_url, body=self.authorize_body)
         return target_url
 
     def test_login(self):
-        self.strategy.set_settings({
-            'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL': True
-        })
-        self.do_login()
+        for environment in VALID_ENVIRONMENTS:
+            with self.subTest():
+                self.strategy.set_settings({
+                    'YLETUNNUS_ENVIRONMENT': environment,
+                    'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL': True
+                })
+                self.do_login()
+
+    def test_login_invalid_environment_raises_error(self):
+        for environment in INVALID_ENVIRONMENTS:
```

Comment:
Is it possible to use pytest's `@pytest.mark.parametrize` here?

Reply:
the subTest call is unittest way to do the same thing.

Reply:
Yeah, I'm not 100% how mixing with pytest would work here, and maybe it's cleaner to use one style in one file/module?

Reply:
> The following pytest features do not work, and probably never > will due to different design philosophies:
>
> - Fixtures (except for autouse fixtures, see below);
> - Parametrization;
> - Custom hooks;

https://docs.pytest.org/en/stable/unittest.html